networkd/network_server.ml

```diff
@@ -336,6 +336,12 @@ module Interface = struct
 			Sysfs.is_physical name
 		) ()
 
+	let has_vlan _ dbg ~name ~vlan =
+		(* Identify the vlan is used by kernel which is unknown to XAPI *)
+		Debug.with_thread_associated dbg (fun () ->
+			List.mem ("", vlan, name) (Proc.get_vlans ())
```

Comment:
Let's not bother with the `full_conf` argument to `get_vlans` and always return the "full" data. You can still ignore the device name like this:

    List.exists (fun (_, v, p) -> v = vlan && p = name) (Proc.get_vlans ())